database.py

```diff
@@ -0,0 +1,55 @@
+from sqlalchemy import create_engine, URL
+from sqlalchemy.orm import Session, sessionmaker
+from sqlalchemy.sql import func
+from functools import cached_property
+from .credentials import cred
+from . import schemas
+from .models.book import Book
+from .logs.logger import logger
+
+
+class Database:
+
+    @cached_property
+    def url_object(self):
+        self.cred = cred.get()
+
+        url_object = URL.create(
```

Comment:
If `url_object` is used only to create an engine - it could be a local variable inside the engine property.